### PR TITLE
[Neo Fix] fix the store crash issue

### PR DESCRIPTION
**Pull request by Jim8y**
# Description

This pr focus on fixing the issue mentioned in https://github.com/neo-project/neo/issues/3123 that the default value of Factory and the initialization order of NeoSystem and StoreFactory caused the null reference.

Fixes # (issue) https://github.com/neo-project/neo/issues/3123

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Run existing tests.

**Test Configuration**:

# Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/Neo/NeoSystem.cs [hunk 1]
```diff
@@ -211,11 +222,12 @@ public void EnsureStopped(IActorRef actor)
         /// <summary>
         /// Loads an <see cref="IStore"/> at the specified path.
         /// </summary>
+        /// <param name="storageEngine">The storage engine used to create the <see cref="IStore"/> objects.</param>
         /// <param name="path">The path of the storage.</param>
         /// <returns>The loaded <see cref="IStore"/>.</returns>
-        public IStore LoadStore(string path)
+        public static IStore LoadStore([NotNull] string storageEngine, string path)
```
Comment (shargon) [inline src/Neo/NeoSystem.cs:228]: Better to remove this method and use `StoreFactory`

File: src/Neo.CLI/CLI/MainService.cs [hunk 2]
```diff
@@ -377,8 +377,7 @@ public async void Start(CommandLineOptions options)
             ProtocolSettings protocol = ProtocolSettings.Load("config.json");
             CustomProtocolSettings(options, protocol);
             CustomApplicationSettings(options, Settings.Default);
-            var store = StoreFactory.GetStore(Settings.Default.Storage.Engine, string.Format(Settings.Default.Storage.Path, protocol.Network.ToString("X8")));
```
Comment (shargon) [inline src/Neo.CLI/CLI/MainService.cs:380]: Use here `?? nameof(MemoryStore)` instead of a new constructor?

**Review by shargon — APPROVED** [review]

File: src/Neo/NeoSystem.cs [hunk 3]
```diff
@@ -208,16 +218,6 @@ public void EnsureStopped(IActorRef actor)
...
-        /// <summary>
-        /// Loads an <see cref="IStore"/> at the specified path.
-        /// </summary>
-        /// <param name="path">The path of the storage.</param>
-        /// <returns>The loaded <see cref="IStore"/>.</returns>
-        public IStore LoadStore(string path)
-        {
-            return StoreFactory.GetStore(store.GetType().Name, path);
-        }
-
```
Comment (cschuchardt88) [inline src/Neo/NeoSystem.cs:220]: why did you remove this? This will break plugins

Comment (cschuchardt88) [reply src/Neo/NeoSystem.cs:220]: Mark as obsolete

Comment (Jim8y) [reply src/Neo/NeoSystem.cs:220]: https://github.com/neo-project/neo/pull/3126 now check it here, wont mark it as absolete, still using it. My bad.

Comment (cschuchardt88) [reply src/Neo/NeoSystem.cs:220]: Its an public `API` should be marked as `obsolete`. If you want to remove it.

Comment (Jim8y) [reply src/Neo/NeoSystem.cs:220]: > We can use Factory in modules, it's not a problem, currently only neo use neo library.

No, I have searched, neoxp also use it, not sure any other tools also using it. @cschuchardt88 is right, its public, cant be changed easily.
